bigtable/client/table.cc

```diff
@@ -152,19 +161,84 @@ RowReader Table::ReadRows(RowSet row_set, std::int64_t rows_limit,
                        bigtable::internal::ReadRowsParserFactory>());
 }
 
-std::pair<bool, Row> Table::ReadRow(std::string row_key, Filter filter) {
+std::pair<bool, Row> Table::ReadRow(std::string row_key, Filter filter,
+                                    grpc::Status& ret_status) {
   RowSet row_set(std::move(row_key));
   std::int64_t const rows_limit = 1;
   RowReader reader =
-      ReadRows(std::move(row_set), rows_limit, std::move(filter));
+      ReadRows(std::move(row_set), rows_limit, std::move(filter), ret_status);
+  if (not ret_status.ok()) {
+    return std::make_pair(false, Row("", {}));
+  }
   auto it = reader.begin();
   if (it == reader.end()) {
+    grpc::Status status = reader.finish();
+    if (not status.ok()) {
+      ret_status = grpc::Status(status.error_code(), status.error_message());
+    }
     return std::make_pair(false, Row("", {}));
   }
   auto result = std::make_pair(true, std::move(*it));
   if (++it != reader.end()) {
-    internal::RaiseRuntimeError(
-        "internal error - RowReader returned 2 rows in ReadRow()");
+    ret_status =
+        grpc::Status(grpc::StatusCode::INTERNAL,
+                     "internal error - RowReader returned 2 rows in ReadRow()");
+    return std::make_pair(false, Row("", {}));
+  }
+  return result;
+}
+
+}  // namespace noex
+
+// Call the `google.bigtable.v2.Bigtable.MutateRow` RPC repeatedly until
+// successful, or until the policies in effect tell us to stop.
+void Table::Apply(SingleRowMutation&& mut) {
+  grpc::Status ret_status = grpc::Status::OK;
```

Comment:
Can you check what is the default constructor for `grpc::Status`  If it initializes to OK we do not need these right?